src/lib/PostgresMetaFunctions.ts

```diff
@@ -57,4 +63,101 @@ export default class PostgresMetaFunctions {
       return { data: null, error: { message: 'Invalid parameters on function retrieve' } }
     }
   }
+
+  async create({
+    name,
+    schema = 'public',
+    params,
+    definition,
+    rettype = 'void',
+    language = 'sql',
+  }: {
+    name: string
+    schema?: string
+    params?: string[]
+    definition: string
+    rettype?: string
+    language?: string
+  }): Promise<PostgresMetaResult<PostgresFunction>> {
+    const sql = `
+      CREATE FUNCTION ${ident(schema)}.${ident(name)}
+      ${params && params.length ? `(${params.join(',')})` : '()'}
+      RETURNS ${rettype || 'void'}
+      AS '${definition}'
+      LANGUAGE ${language}
+      RETURNS NULL ON NULL INPUT;
+      `
+    const { error } = await this.query(sql)
+    if (error) {
+      return { data: null, error }
+    }
+    return await this.retrieve({ name, schema })
+  }
+
+  async update(
+    id: number,
+    {
+      name,
+      schema = 'public',
+      params,
+      extension,
+    }: {
+      name: string
+      schema?: string
+      params?: string[] //optional params for overloaded functions
+      extension?: string //e.g. sqrt DEPENDS ON EXTENSION mathlib
+    }
+  ): Promise<PostgresMetaResult<PostgresFunction>> {
+    const { data: old, error: retrieveError } = await this.retrieve({ id })
+    if (retrieveError) {
+      return { data: null, error: retrieveError }
+    }
+
+    let alter = `ALTER FUNCTION ${ident(old!.schema)}.${ident(old!.name)}${
+      params && params.length ? `(${params.join(',')})` : ''
+    }`
+
+    let schemaSql = ''
+    if (schema !== undefined && schema !== old!.schema) {
+      schemaSql = `${alter} SET SCHEMA ${ident(schema)};`
+      alter = `ALTER FUNCTION ${ident(schema)}.${ident(old!.name)}${
+        params && params.length ? `(${params.join(',')})` : ''
+      }`
```

Comment:
@w3b6x9  Thanks again for the great feedback. So far I've made all the changes except this one (having a little trouble here). 

Even with the order of `schemaSql` and `nameSql` reversed, I still have to reset `alter` because when I reverse `schemaSql` and `nameSql`, by the time `schemaSql` is ran (after `nameSql`), `ALTER FUNCTION ${ident(old!.name)}` would no longer be valid because `old!.name` would have been replace by the newly provided `name`  

What I've done instead is dropped the if statement and kept your suggestion (swapped the order of `nameSql` & `schemaSql`) but still resetting `alter`